pkg/cmd/package/upload/upload.go

```diff
@@ -0,0 +1,182 @@
+package upload
+
+import (
+	"errors"
+	"fmt"
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/OctopusDeploy/cli/pkg/constants"
+	"github.com/OctopusDeploy/cli/pkg/constants/annotations"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/OctopusDeploy/cli/pkg/util"
+	"github.com/OctopusDeploy/cli/pkg/util/flag"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/newclient"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/packages"
+	"github.com/spf13/cobra"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// TODO Outstanding questions for package upload:
+// - What should the response be for outputformat basic and json?
```

Comment:
The basic output from `octo` looks something like this:

```
Octopus CLI, version 9.x.y

Detected automation environment: "NoneOrUnknown"
Space name unspecified, process will run in the default space context
Handshaking with Octopus Server: https://foobar.octopus.app
Handshake successful. Octopus version: 2022.x.y; API version: 3.0.0
Authenticated as: Foo Bar <foo@bar.com> 
Pushing package: /Users/foobar/foo.1.1.0.zip...
Requesting signature for delta compression from the server for upload of a package with id 'foo' and version '1.1.0'
No package with the same ID exists on the server
Falling back to pushing the complete package to the server
Push successful
```

Here's the output in JSON:

```
{
  "SuccessfulPackages": [
    "/Users/johnbristowe/GitHub/OctopusCLI/foo.1.1.1.zip"
  ],
  "FailedPackages": []
}
```

Reply:
My intent with all these TODOS was to merge this PR first as a 'first pass' that people could play with, then do a second pass and tidy them all up, plus any other UX/wording feedback we gathered.

However, I didn't actually tell anyone about that. Whoops, so of course you'd reject the PR on grounds of unresolved TODOs :-)